### PR TITLE
refactor: user friendly bouncer new swap cmd 

### PR DESCRIPTION
**Pull request by j4m1ef0rd**
# Pull Request

Closes: PRO-1559 & PRO-1560
## Checklist

Please conduct a thorough self-review before opening the PR.

- [x] I am confident that the code works.
- [ ] I have written sufficient tests.
- [ ] I have written and tested required migrations.
- [ ] I have updated documentation where appropriate.

## Summary

- New swap cmd now takes a user friendly price in base_asset per quote_asset, eg Dot/Btc. 100 = 100 dot/btc
- New swap cmd can handle different types of DOT address on the refund address.

### CODE REVIEW

File: bouncer/commands/new_swap.ts [hunk 1]
```diff
@@ -5,12 +5,17 @@
 // <sourceAsset> <destAsset> <destAddress> [maxBoostFeeBps] [refundAddress] [minPrice] [refundDuration]
 // Use `-h` for help.
 // If the refundAddress is provided, the minPrice must also be provided. The refundDuration will default to 0 if not provided.
-// Example: ./commands/new_swap.ts Dot Btc n1ocq2FF95qopwbEsjUTy3ZrawwXDJ6UsX --refundAddress "0xa0b52be60216f8e0f2eb5bd17fa3c66908cc1652f3080a90d3ab20b2d352b610" --minPrice 100000000000000000
+// Example: ./commands/new_swap.ts Dot Btc n1ocq2FF95qopwbEsjUTy3ZrawwXDJ6UsX --refundAddress "0xa0b52be60216f8e0f2eb5bd17fa3c66908cc1652f3080a90d3ab20b2d352b610" --minPrice 100
```
Comment (kylezs) [inline bouncer/commands/new_swap.ts:8]: Be nice to explain here what this price means

File: bouncer/commands/new_swap.ts [hunk 2]
```diff
@@ -77,8 +82,11 @@ async function newSwapCommand() {
     args.refundAddress !== undefined && args.minPrice !== undefined
       ? {
           retryDurationBlocks: args.refundDuration,
-          refundAddress: args.refundAddress,
-          minPrice: args.minPrice,
+          refundAddress:
+            args.sourceAsset === 'Dot'
+              ? decodeDotAddressForContract(args.refundAddress)
+              : args.refundAddress,
+          minPrice: usdcToInternalPrice(args.sourceAsset as InternalAsset, args.minPrice),
```
Comment (kylezs) [inline bouncer/commands/new_swap.ts:89]: the example has Dot -> Btc, how is the price represented in that case?

Comment (j4m1ef0rd) [reply bouncer/commands/new_swap.ts:89]: The price you would enter is the min price of DOT in USDC. say 4, because DOT is worth $4.15 right now.

Comment (kylezs) [reply bouncer/commands/new_swap.ts:90]: but what about the price of bitcoin?

Comment (j4m1ef0rd) [reply bouncer/commands/new_swap.ts:90]: oh sorry. I misunderstood what price is. A little confusing 😅.

Comment (j4m1ef0rd) [reply bouncer/commands/new_swap.ts]: Ok i think i got it correct now. The "price" is in Dot per Btc and then converts to dot_planck/Satoshi taking into account the decimals of both assets. This internally works out to a minimum output in Satoshi. I checked the minimum output to confirm it was what i expected.

**Review by kylezs — APPROVED** [review]

**Pull request merged**